### PR TITLE
[layer] Update LSTM for layer_v2 

### PR DESCRIPTION
**Pull request by kparichay**
Update LSTM for layer_v2.
Also enable corresonding unittests and add common unittests.

Update LSTM implementation to reduce the use of temporary tensors
and request temporary tensors for the ones that are necessary.

Signed-off-by: Parichay Kapoor <kparichay@gmail.com>

### CODE REVIEW

**Review by zhoonit — APPROVED** [review]
LGTM

File: nntrainer/layers/lstm.cpp [hunk 1]
```diff
@@ -228,91 +233,72 @@ void LSTMLayer::forwarding(bool training) {
       acti_func.run_fn(cs, hs);
       hs.multiply_i(ho);
     }
-    // size of h_prev and hs size is same : unit.
-    // size of c_prev and cs is same : unit.
-    h_prev.getBatchSlice(b, 1).copy(hs);
-    c_prev.getBatchSlice(b, 1).copy(cs);
```
Comment (jijoongmoon) [inline nntrainer/layers/lstm.cpp:234]: I wonder if it pass the unittest?

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/layers/lstm.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/layers/lstm.cpp:234]: Yes, the golden unittest of RNN is in unittest_models which has been activated with this PR.
From what I understood, h_prev was used to cache the values of the current calculation for the next hidden state, and then the next hidden state was initialized with it. 
So, I updated to cache directly in the memory of the next hidden state directly.

Comment (jijoongmoon) [reply nntrainer/layers/lstm.cpp:234]: great, thanks.

**Pull request merged**